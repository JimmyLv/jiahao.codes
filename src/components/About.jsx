/* eslint-disable react/no-danger */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import { GitHub, Keybase, LinkedIn, Twitter } from './svgs';
import { siteSocial } from '../proptypes';

const SocialLinks = styled.div`
  text-align: right;
`;

const SocialLink = styled.a`
  display: inline-block;
  background-image: none;

  &:hover {
    background-image: none;
  }
`;

export default function About({
  title,
  html,
  social: { githubUrl, keybaseUrl, linkedInUrl, twitterUrl },
}) {
  return (
    <div>
      <h1>{title}</h1>
      <div dangerouslySetInnerHTML={{ __html: html }} />
      <SocialLinks>
        <SocialLink href={githubUrl} target="_blank" rel="noopener noreferrer">
          <GitHub enlargeOnMobile />
        </SocialLink>
        <SocialLink href={keybaseUrl} target="_blank" rel="noopener noreferrer">
          <Keybase enlargeOnMobile />
        </SocialLink>
        <SocialLink href={twitterUrl} target="_blank" rel="noopener noreferrer">
          <Twitter enlargeOnMobile />
        </SocialLink>
        <SocialLink
          href={linkedInUrl}
          target="_blank"
          rel="noopener noreferrer"
          enlargeOnMobile
        >
          <LinkedIn />
        </SocialLink>
      </SocialLinks>
    </div>
  );
}

About.propTypes = {
  html: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  social: siteSocial.isRequired,
};
