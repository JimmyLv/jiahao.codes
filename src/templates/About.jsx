import React from 'react';
import PropTypes from 'prop-types';

import { Head, About } from '../components';
import {
  markdownRemark as markdownRemarkPropType,
  site as sitePropType,
} from '../proptypes';

export default function AboutTemplate({
  data: { site, markdownRemark: { frontmatter: { title, path }, html } },
}) {
  return (
    <div>
      <Head title={title} path={path} site={site} />
      <About title={title} html={html} social={site.siteMetadata.social} />
    </div>
  );
}

AboutTemplate.propTypes = {
  data: PropTypes.shape({
    markdownRemark: markdownRemarkPropType,
    site: sitePropType.isRequired,
  }).isRequired,
};

export const pageQuery = graphql`
  query AboutPage($path: String!) {
    site {
      siteMetadata {
        siteUrl
        title
        author
        description
        facebookAppId
        twitterUser
        social {
          githubUrl
          twitterUrl
          keybaseUrl
        }
      }
    }
    markdownRemark(frontmatter: { path: { eq: $path } }) {
      id
      html
      excerpt
      frontmatter {
        path
        title
        excerpt
        date(formatString: "MMMM DD, YYYY")
      }
    }
  }
`;
