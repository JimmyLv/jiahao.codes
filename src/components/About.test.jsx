import React from 'react';
import { shallow } from 'enzyme';

import About from './About';

test('rendering', () => {
  const props = {
    title: 'title',
    html: '<div>hello</div>',
    social: {
      githubUrl: 'githubUrl',
      keybaseUrl: 'keybaseUrl',
      linkedInUrl: 'linkedInUrl',
      twitterUrl: 'twitterUrl',
    },
  };
  const wrapper = shallow(<About {...props} />);
  expect(wrapper).toMatchSnapshot();
});
