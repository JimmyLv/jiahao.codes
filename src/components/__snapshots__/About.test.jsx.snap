// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <h1>
    title
  </h1>
  <div
    dangerouslySetInnerHTML={
      Object {
        "__html": "<div>hello</div>",
      }
    }
  />
  <styled.div>
    <styled.a
      href="githubUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My GitHub Profile"
    >
      <GithubIcon
        enlargeOnMobile={true}
      />
    </styled.a>
    <styled.a
      href="keybaseUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My Keybase Profile"
    >
      <KeybaseIcon
        enlargeOnMobile={true}
      />
    </styled.a>
    <styled.a
      href="twitterUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My Twitter Profile"
    >
      <TwitterIcon
        enlargeOnMobile={true}
      />
    </styled.a>
    <styled.a
      enlargeOnMobile={true}
      href="linkedInUrl"
      rel="noopener noreferrer"
      target="_blank"
      title="My LinkedIn Profile"
    >
      <LinkedIn />
    </styled.a>
  </styled.div>
</div>
`;
